### PR TITLE
fix: setuptools is unbundled on Gentoo

### LINKED ISSUE #1863: Please provide support for non-setuptools.extern bundled packages
**Describe the bug**
The package currently uses `packaging`, `importlib_metadata` and `tomli` bundled with setuptools via `setuptools.extern` package. Setuptools is unbundled on Gentoo, therefore `setuptools.extern` is not provided and cx_Freeze fails due to missing imports, e.g.:

```pytb
____________________________________________ ERROR collecting tests/test_winversioninfo.py ____________________________________________
ImportError while importing test module '/tmp/cx_Freeze/tests/test_winversioninfo.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_winversioninfo.py:5: in <module>
    from cx_Freeze.winversioninfo import Version, VersionInfo
cx_Freeze/winversioninfo.py:12: in <module>
    from setuptools.extern.packaging import version as packaging_version
E   ModuleNotFoundError: No module named 'setuptools.extern'
```

Could you please add support for falling back to the top-level `packaging`, `tomli` and `importlib_metadata`, please?

**To Reproduce**
```
git clone https://github.com/marcelotduarte/cx_Freeze
cd cx_Freeze
python -m venv .venv
. .venv/bin/activate
pip install -r requirements-dev.txt

# this is equivalent to Gentoo's unvendoring
pip install jaraco.text more-itertools nspektr ordered-set packaging platformdirs tomli importlib_metadata
rm -r .venv/lib/python3.*/site-packages/setuptools/{extern,_vendor}
find .venv/lib/python3.*/site-packages/setuptools/ -name '*.py' -exec sed -e 's:from \w*[.]\+extern ::' -e 's:\w*[.]\+extern[.]::' -i {} +

python -m pytest
```

**Expected behavior**
Tests passing ;-).

**Desktop (please complete the following information):**
 - Platform information (e.g. Ubuntu Linux 22.04): Gentoo Linux
 - OS architecture (e.g. amd64): amd64
 - cx_Freeze version [e.g. 6.11]: 6.14.8, 2642a2579f355fea2a732e152ca762bd5ad9c98c
 - Python version [e.g. 3.10]: 3.11.2

### PR DESCRIPTION
**Pull request by marcelotduarte**
Fixes #1863

### CODE REVIEW

File: ci/requirements.py [hunk 1]
```diff
@@ -6,9 +6,12 @@
 from pathlib import Path
 
 try:
-    import tomllib
```
Comment (mgorny) [inline ci/requirements.py:9]: For the record, you could also do `import tomli as tomllib` if you preferred to keep the original style.

**Review by mgorny — COMMENTED** [review]
Thank you. I've applied the patch on top of 6.14.8 and I've confirmed that it fixes the import errors on Gentoo. I still get two test failures that I suspect are related to newer setuptools version but that's a separate problem (and possibly one already fixed in git, I haven't checked).

File: ci/requirements.py (hunk 1 above)
Comment (marcelotduarte) [reply ci/requirements.py:9]: Absolutely, but I saw it somewhere in conda using toml_load and took the opportunity to change it.

**Comment by marcelotduarte** [comment]
@mgorny On top of 6.14.8 tests/test_winversioninfo.py should be fixed. Then the tests passed with warnings.
```

============================================================= warnings summary =============================================================
../../.local/venv/cx310/lib/python3.10/site-packages/pkg_resources/__init__.py:121
  /home/marcelo/.local/venv/cx310/lib/python3.10/site-packages/pkg_resources/__init__.py:121: DeprecationWarning: pkg_resources is deprecated as an API
    warnings.warn("pkg_resources is deprecated as an API", DeprecationWarning)

../../.local/venv/cx310/lib/python3.10/site-packages/pkg_resources/__init__.py:2870
../../.local/venv/cx310/lib/python3.10/site-packages/pkg_resources/__init__.py:2870
../../.local/venv/cx310/lib/python3.10/site-packages/pkg_resources/__init__.py:2870
../../.local/venv/cx310/lib/python3.10/site-packages/pkg_resources/__init__.py:2870
  /home/marcelo/.local/venv/cx310/lib/python3.10/site-packages/pkg_resources/__init__.py:2870: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================= short test summary info ==========================================================
SKIPPED [1] tests/test_command_bdist_msi.py:25: Windows tests
SKIPPED [1] tests/test_command_bdist_msi.py:44: Windows tests
SKIPPED [1] tests/test_command_bdist_msi.py:57: Windows tests
SKIPPED [1] tests/test_command_bdist_msi.py:70: Windows tests
SKIPPED [1] tests/test_winmsvcr.py:12: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:17: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:34: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:44: Windows tests
SKIPPED [1] tests/samples/plist_items/test_plist_items.py:14: Macos tests
================================================ 13 passed, 9 skipped, 5 warnings in 21.06s ================================================
```
The warning are removed in main branch.

**Comment by marcelotduarte** [comment]
The same test on top of main:
https://github.com/marcelotduarte/cx_Freeze/actions/runs/4622726728/jobs/8175758568?pr=1864

**Comment by mgorny** [comment]
What I'm seeing is:

```
FAILED tests/test_command_build.py::test_build - TypeError: ConfigError.__init__() missing 1 required positional argument: 'msg'
FAILED tests/test_command_build_exe.py::test_build_exe - TypeError: ConfigError.__init__() missing 1 required positional argument: 'msg'
```

Full log (1.7M): 
[dev-python:cx_Freeze-6.14.8:20230406-141948.log](https://github.com/marcelotduarte/cx_Freeze/files/11173466/dev-python.cx_Freeze-6.14.8.20230406-141948.log)

**Comment by marcelotduarte** [comment]
This two errors occurs because cx_Freeze is not installed. Run this command before pytest:
`pip install -e . --no-build-isolation --no-deps`

I tested with docker image gentoo/stage3

To test the bdist_rpm install rpmbuild:
`emerge --ask app-arch/rpm`

Wow, the are no binaries on Gentoo. Compilation of source code all the time ;-)

**Comment by mgorny** [comment]
Thanks. We're running the test suite as part of the ebuild, i.e. as part of `emerge -v dev-python/cx_Freeze` — but I'll recheck and investigate once the next release is out.

**Comment by mgorny** [comment]
Thanks, again. It works now ;-).

I have also figured out our test failure — pytest was importing `cx_Freeze` from the current directory instead of the installed version.

**Comment by marcelotduarte** [comment]
Good. You're welcome.

**Pull request merged**